rust_src/src/multibyte.rs

```diff
@@ -279,6 +279,25 @@ impl LispObject {
     pub fn empty_unibyte_string() -> LispStringRef {
         LispStringRef::from(unsafe { empty_unibyte_string })
     }
+
+    // Creates a LispObject on stack representing the unibyte string `s`
```

Comment:
How is this "on the stack"? https://doc.rust-lang.org/rust-by-example/std/box.html is clear that a Box value is a heap value. All heap must be C managed currently.

This code calls `Box<T>::into_raw` which passes ownership to the C code. However, the C code has not put this pointer into the garbage collector. Which means this is likely a memory leak.

The mock function was to emulate the creation of strings for small tests. If this really worked correctly we would be writing Rust unit tests.

To accept this code we need to be certain that the garbage collector will properly handle the cleanup. Is the call to `LispObject::tag_ptr` sufficient? I am not certain that it is.

Reply:
I didn't realize this. Will correct this.

Reply:
Even the memory used by `strcopy` was leaking.

Reply:
See the changes now. If these are fine, we can replace usage of `mock_unibyte_string` with the new `local_unibyte_string` too.